src/lib/ProductImporter.ts

```diff
@@ -462,7 +465,8 @@ export class ProductImporter {
 
         let productCharges = [...this.extractProductObjects(allCharges, this.chargesIds)];
     
-        productCharges.forEach(charge=>{this.charges.push(charge['root']);
+        productCharges.forEach(charge=>{if(charge['root']['enxCPQ__Charge_Reference__r']){ this.charges.push(charge['root'])}
```

Comment:
For me, the formatting here is a little bit dirty and hard to read. I would split it to multiple lines. And I would also consider using filter() and map() functions instead of push(). For example something like that:
```
this.charges = {...this.charges, productCharges
    .filter(charge => charge['root']['enxCPQ__Charge_Reference__r'] )
    .map(charge => charge['root'])
};
```